controllers/solrprometheusexporter_controller.go

```diff
@@ -92,10 +124,18 @@ func (r *SolrPrometheusExporterReconciler) Reconcile(req ctrl.Request) (ctrl.Res
 		if err != nil && errors.IsNotFound(err) {
 			configMapLogger.Info("Creating ConfigMap")
 			err = r.Create(context.TODO(), configMap)
+
+			// capture the MD5 for the config XML
+			configXml, _ := configMap.Data[configMapKey]
+			configXmlMd5 = fmt.Sprintf("%x", md5.Sum([]byte(configXml)))
```

Comment:
You need to move this before `if/else`, otherwise the xml annotation will be removed when the reconcile is called, but the configMap is not updated or created.

See: https://github.com/apache/lucene-solr-operator/pull/188/files#diff-27cb619f80d28ac158f7885b7925d8582babfa7fe1f824599d65a745302d237cR222

Reply:
ah yes ... great catch, and thanks for fixing for solr.xml